src/DOM/HTML/Window.purs

```diff
@@ -102,3 +109,31 @@ foreign import scrollY :: forall eff. Window -> Eff (dom :: DOM | eff) Int
 foreign import localStorage :: forall eff. Window -> Eff (dom :: DOM | eff) Storage
 
 foreign import sessionStorage :: forall eff. Window -> Eff (dom :: DOM | eff) Storage
+
+newtype RequestAnimationFrameId = RequestAnimationFrameId Int
+
+derive instance newtypeRequestAnimationFrameId :: Newtype RequestAnimationFrameId _
+
+foreign import _requestAnimationFrame :: forall eff. Eff (dom :: DOM | eff) Unit -> Window -> Eff (dom :: DOM | eff) Int
+
+requestAnimationFrame :: forall eff. Eff (dom :: DOM | eff) Unit -> Window -> Eff (dom :: DOM | eff ) RequestAnimationFrameId
+requestAnimationFrame fn = map RequestAnimationFrameId <<< _requestAnimationFrame fn
+
+foreign import _cancelAnimationFrame :: forall eff. Int -> Window -> Eff (dom :: DOM | eff) Unit
+
+cancelAnimationFrame :: forall eff. RequestAnimationFrameId -> Window -> Eff (dom :: DOM | eff) Unit
+cancelAnimationFrame idAF = _cancelAnimationFrame (unwrap idAF)
+
+newtype RequestIdleCallbackId = RequestIdleCallbackId Int
+
+derive instance newtypeRequestIdleCallbackId :: Newtype RequestIdleCallbackId _
+
+foreign import _requestIdleCallback :: forall eff. Eff (dom :: DOM | eff) Unit -> { timeout :: Int } -> Window -> Eff (dom :: DOM | eff) Int 
+
+requestIdleCallback :: forall eff. Eff (dom :: DOM | eff) Unit -> { timeout :: Int } -> Window -> Eff (dom :: DOM | eff ) RequestIdleCallbackId
```

Comment:
The spec says that the timeout is only taken into account if it is a [positive integer](https://w3c.github.io/requestidlecallback/#h-the-requestidlecallback-method).